index.html

```diff
@@ -75,29 +75,25 @@ <h3><a id="developers" href="#developers">Don’t think, just strip</a></h3>
           <p>Here is a sample PHP function for developers:</p>
 
           <pre><code>&lt;?php
-function is_naked_day($d) {
-  $start = date('U', mktime(-14, 0, 0, 04, $d, date('Y')));
-  $end = date('U', mktime(36, 0, 0, 04, $d, date('Y')));
-  $z = date('Z') * -1;
-  $now = time() + $z;
-  if ( $now &gt;= $start &amp;&amp; $now &lt;= $end ) {
-    return true;
-  }
-  return false;
+function isNakedDay()
+{
+    $now = new \DateTime('now', new \DateTimeZone('UTC'));
+    $start = new \DateTime('00:00, April 9', new \DateTimeZone('+14:00'));
+    $end = new \DateTime('23:59:59.999999, April 9', new \DateTimeZone('-12:00'));
+
+    return $now &gt;= $start && $now &lt;= $end;
```

Comment:
Should we simplify it a step further? :)

This would also ensure consistency with the explanation below the code.

```suggestion
function isNakedDay()
{
    $now = new \DateTime('now', new \DateTimeZone('UTC'));
    $start = new \DateTime('00:00, April 9', new \DateTimeZone('+14:00'));
    $end = new \DateTime('00:00, April 10', new \DateTimeZone('-12:00'));

    return $now &gt;= $start && $now &lt; $end;
```

Reply:
The term _Midnight_ is ambiguous, indeed, as are 12-hours clocks, with _AM_ and _PM_ being used to try to make some sense out of it. I know of no confusion about _00:00_ being the start of the day though.

As the article you’re pointing to tells us, and based on ISO, while they represent the exact same time, _00:00:00_ is the start of the day, _24:00:00_ the end of the day, and _Midnight_ subject to interpretations. :)

I wonder how a `\DateTime('24:00, April 9', new \DateTimeZone('-12:00'));` would behave.